### DISCUSSION TITLE
Error sending SMTP test email SmtpError

### DISCUSSION THREAD

**Discussion by rpelissi** [dani-garcia/vaultwarden#1662] [Q&A]
### Subject of the issue
Can't send emails even after configuring basic smtp settings

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.21.0
* Web-vault version: v2.19.0d
* Running within Docker: false
* Uses a reverse proxy: false
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.33.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**.**********.***",
  "domain_origin": "*****://**.**********.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Bitwarden_RS",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": true,
  "smtp_explicit_tls": false,
  "smtp_from": "**@**********.***",
  "smtp_from_name": "Bitwarden_RS",
  "smtp_host": "***.***.*.*",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

### Steps to reproduce
I have vaultwarden installed as a docker image into a k3s/rancher environment. I have setup smtp settings (very simple since I use internal smtp server to forward email to the outside) but everytime I try the test email, it fails with the error message:
`Error sending SMTP test email SmtpError`

And in the docker logs, I see:
```
[2021-05-03 20:05:47.368][error][ERROR] SmtpError.

[CAUSE] lettre::transport::smtp::Error {

    kind: Connection,

    source: "Could not connect",

}
```

### Expected behaviour
I should be able at least to see the connection attemps on my internal smtp postfix setup.

### Actual behaviour
Error message when testing emails.

**Comment by BlackDex** [comment]
Well, you can only see an attempt if it was able to connect, which it isn't. So either the IP, or the port is incorrect, or there is a firewall which blocks something, or that container isn't allowed to to connect to that internal ip.

**Reply by rpelissi** [reply]
You're perfectly right, sorry for this bug report. I got an issue with my mail relay. Thanks!

**Reply by Lowxorx** [reply]
> You're perfectly right, sorry for this bug report. I got an issue with my mail relay. Thanks!

Hello, 
I have the same problem with my postfix smtp relay, I would like to know what you did to solve your error? Thanks in advance.
